flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/TimestampedCollector.java

```diff
@@ -71,4 +73,19 @@ public void eraseTimestamp() {
 	public void close() {
 		output.close();
 	}
+
+	@Override
+	public void emitWatermark(Watermark mark) {
+		output.emitWatermark(mark);
+	}
+
+	@Override
+	public <X> void collect(OutputTag<X> outputTag, StreamRecord<X> record) {
```

Comment:
It is involved in three new methods by implementing `Output` interface instead. But actually we only need the `emitWatermark` method, and the left two methods would never be touched in codes. If so, another option is that we can only define `emitWatermark` method here and still implement `Collector` interface.

Reply:
Yes, ideally we move all `emit*` to the `TimestampedCollector` and avoid going through the unwrapped output. I'm checking how much code would be affected.

Reply:
Opening a separate ticket for that as many code paths are touched.